langchain4j-mongodb-atlas/src/main/java/dev/langchain4j/store/embedding/mongodb/MongoDbMetadataFilterMapper.java

```diff
@@ -0,0 +1,95 @@
+package dev.langchain4j.store.embedding.mongodb;
+
+import com.mongodb.client.model.Filters;
+import dev.langchain4j.store.embedding.filter.Filter;
+import dev.langchain4j.store.embedding.filter.comparison.IsEqualTo;
+import dev.langchain4j.store.embedding.filter.comparison.IsGreaterThan;
+import dev.langchain4j.store.embedding.filter.comparison.IsGreaterThanOrEqualTo;
+import dev.langchain4j.store.embedding.filter.comparison.IsIn;
+import dev.langchain4j.store.embedding.filter.comparison.IsLessThan;
+import dev.langchain4j.store.embedding.filter.comparison.IsLessThanOrEqualTo;
+import dev.langchain4j.store.embedding.filter.comparison.IsNotEqualTo;
+import dev.langchain4j.store.embedding.filter.comparison.IsNotIn;
+import dev.langchain4j.store.embedding.filter.logical.And;
+import dev.langchain4j.store.embedding.filter.logical.Not;
+import dev.langchain4j.store.embedding.filter.logical.Or;
+import org.bson.conversions.Bson;
+
+class MongoDbMetadataFilterMapper {
+
+    public static Bson map(Filter filter) {
+        if (filter instanceof IsEqualTo) {
+            return mapEqual((IsEqualTo) filter);
+        } else if (filter instanceof IsNotEqualTo) {
+            return mapNotEqual((IsNotEqualTo) filter);
+        } else if (filter instanceof IsGreaterThan) {
+            return mapGreaterThan((IsGreaterThan) filter);
+        } else if (filter instanceof IsGreaterThanOrEqualTo) {
+            return mapGreaterThanOrEqual((IsGreaterThanOrEqualTo) filter);
+        } else if (filter instanceof IsLessThan) {
+            return mapLessThan((IsLessThan) filter);
+        } else if (filter instanceof IsLessThanOrEqualTo) {
+            return mapLessThanOrEqual((IsLessThanOrEqualTo) filter);
+        } else if (filter instanceof IsIn) {
+            return mapIn((IsIn) filter);
+        } else if (filter instanceof IsNotIn) {
+            return mapNotIn((IsNotIn) filter);
+        } else if (filter instanceof And) {
+            return mapAnd((And) filter);
+        } else if (filter instanceof Or) {
+            return mapOr((Or) filter);
+        } else if (filter instanceof Not) {
+            return mapNot((Not) filter);
+        } else {
+            throw new UnsupportedOperationException("Unsupported filter type: " + filter.getClass().getName());
+        }
+    }
```

Comment:
This is great. We do all our MQL as dictionaries with a ton of strings.

Reply:
I believe this is referring to the Python MongoDB driver, which @caseyclements is a developer on.